src/User.ts

```diff
@@ -58,6 +54,29 @@ export class User {
     }
 
     const userIdentifier = Utils.base64EncodeStringUrlSafe(uid) + "." + Utils.base64EncodeStringUrlSafe(utid);
-    return new User(idToken.preferredName, idToken.name, idToken.issuer, userIdentifier);
+    return new User(idToken, userIdentifier);
+  }
+
+  get displayableId(): string {
+    return this.idToken.preferredName;
+  }
+
+  get name(): string {
+    return this.idToken.name;
+  }
+
+  get identityProvider(): string {
+    return this.idToken.issuer;
+  }
+
+  getClaim(claim: string): string {
```

Comment:
@davidmorissette The idToken passed in the constructor of user object is already decoded. There is no need to decode it again.